api/resolver/resolver.go

```diff
@@ -11,6 +11,12 @@ var (
 	ErrInvalidPath = errors.New("invalid path")
 )
 
+// NamespaceResolver resolves request to the namespace
+type NamespaceResolver interface {
```

Comment:
Why not just add Namespace to the Endpoint thats resolved in the current resolver?